stablehlo/dialect/Base.h

```diff
@@ -237,9 +248,10 @@ class CompatibleOperandsAndResultType
           location,
           "Expected non-empty operands for [CompatibleOperandsAndResultType]");
 
-    if (failed(inferMostSpecificType(location, operands.getTypes(),
-                                     inferredReturnTypes)))
-      return failure();
+    auto inferredTypeOrErr =
+        inferMostSpecificType(location, operands.getTypes());
+    if (failed(inferredTypeOrErr)) return failure();
+    inferredReturnTypes.emplace_back((*inferredTypeOrErr).cast<ShapedType>());
```

Comment:
I don't think the cast is needed here, because `*inferredTypeOrErr` is already `ShapedType`.

Reply:
It was intended to be moved. Fixed.